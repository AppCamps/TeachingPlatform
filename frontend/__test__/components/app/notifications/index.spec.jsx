import React from 'react';
import { shallow } from 'enzyme';

import Notifications from '../../../../components/app/notifications';

describe('<Notifcations />', () => {
  it('renders notifications', () => {
    const activeNotifications = [
      {
        type: 'failure',
        text: 'Something failed.',
        key: 1234567,
      },
      {
        type: 'success',
        text: 'Yay!',
        key: 7654321,
      },
    ];

    const notifTree = shallow(<Notifications activeNotifications={activeNotifications} />);
    global.expect(notifTree).toMatchSnapshot();
  });

  it('renders empty state', () => {
    const activeNotifications = [];

    const notifTree = shallow(<Notifications activeNotifications={activeNotifications} />);
    global.expect(notifTree).toMatchSnapshot();
  });
});
