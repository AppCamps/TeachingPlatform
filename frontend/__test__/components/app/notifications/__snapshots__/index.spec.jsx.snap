// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notifcations /> renders empty state 1`] = `
<div
  className="notificationsContainer"
>
  <div
    className="notifications"
  >
    <div
      className="absolution"
    />
  </div>
</div>
`;

exports[`<Notifcations /> renders notifications 1`] = `
<div
  className="notificationsContainer"
>
  <div
    className="notifications"
  >
    <div
      className="absolution"
    >
      <div
        className="notification failure"
        key="1234567"
      >
        Something failed.
      </div>
      <div
        className="notification success"
        key="7654321"
      >
        Yay!
      </div>
    </div>
  </div>
</div>
`;
